buildSrc/src/main/java/quilt/internal/tasks/lint/SpellingChecker.java

```diff
@@ -0,0 +1,190 @@
+import cuchaz.enigma.translation.representation.AccessFlags;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+
+public class SpellingChecker implements Checker<Entry<?>> {
+    private static final String PACKAGE_INFO_CLASS_PACKAGE = "net/minecraft/unused/packageinfo/";
+    private static final String METHOD_PREFIX = "m_";
+    private static final String CLASS_PREFIX = "C_";
+    private static final char PACKAGE_SEPARATOR = '/';
+
+    private static final Set<String> MINECRAFT_WORDS = new HashSet<>();
+    static {
+        // collect minecraft words
+        InputStream parsedFile = SpellingChecker.class.getClassLoader().getResourceAsStream("minecraft_specific_words.txt");
+        if (parsedFile == null) {
+            throw new RuntimeException("could not find minecraft word file for spelling check in mapping lint task!");
+        }
+
+        Set<String> minecraftWords = getLines(parsedFile);
+        MINECRAFT_WORDS.addAll(minecraftWords);
+    }
+
+    private static Set<String> getLines(InputStream stream) {
+        return new BufferedReader(new InputStreamReader(stream)).lines().filter(line -> !line.isBlank() && !line.startsWith("//")).collect(Collectors.toSet());
+    }
+
+    private static final Set<String> words = new HashSet<>();
+
+    @Override
+    public void update(MappingLintTask.LintParameters parameters) {
+        words.addAll(MINECRAFT_WORDS);
+
+        try {
+            FileInputStream spellingFile = new FileInputStream(parameters.getSpellingFile().get().getAsFile());
+            Set<String> spellingWords = getLines(spellingFile);
+            words.addAll(spellingWords);
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Unable to find spelling file!");
+        }
+    }
+
+    @Override
+    public void check(Entry<?> entry, EntryMapping mapping, Function<Entry<?>, AccessFlags> accessProvider, ErrorReporter errorReporter) {
+        String name = mapping.targetName();
+
+        if (name != null
+            && !name.startsWith(PACKAGE_INFO_CLASS_PACKAGE)
+            // ignore unmapped methods and classes
+            && !(entry instanceof MethodEntry && name.startsWith(METHOD_PREFIX))
+            && !(entry instanceof ClassEntry && name.startsWith(CLASS_PREFIX))) {
+
+            checkMapping(entry, name, errorReporter);
+            checkJavadoc(mapping.javadoc(), errorReporter);
+        }
+    }
+
+    private static void checkJavadoc(String javadoc, ErrorReporter errorReporter) {
+        if (javadoc != null && !javadoc.isEmpty()) {
+            // in this case we just have to run the world's slowest regex, there's no clever optimisations based on type to be had
+            List<String> splitJavadoc = splitWords(List.of(javadoc), word -> true, "[0-9_\\[\\]\\-\\.\\{\\}\\(\\)@#/\":; ,<>\n=\\*\\|\\+%`§&!]");
```

Comment:
Why isn't this just `"[^A-Za-z]"`? The current regex also has a lot of redundant escapes, so in reality it looks like this ```"[0-9_\[\]\-.{}()@#/":; ,<>\n=*|+%`§&!]"```

Reply:
that could definitely be changed
one thing to watch for is that I wanted to avoid splitting by apostrophe, so I'll add that to the new one